src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -150,6 +152,40 @@ def _construct_line_result(self, output, run_info):
             line_result.output[LINE_NUMBER_KEY] = run_info.index
         return line_result
 
+    @property
+    def has_aggregation_node(self):
+        return hasattr(self, "_aggr_func")
+
+    def exec_aggregation(
+        self,
+        inputs: Mapping[str, Any],
+        aggregation_inputs: List[Any],
+        run_id: Optional[str] = None,
+    ) -> AggregationResult:
+        if not self._aggr_func:
+            return AggregationResult({}, {}, {})
+        # Similar to dag flow, add a prefix "reduce" for run id of aggregation function.
+        run_id = f"{run_id}_reduce" if run_id is not None else f"{str(uuid.uuid4())}_reduce"
+        with self._update_operation_context_for_aggregation(run_id):
+            return self._exec_aggregation(aggregation_inputs)
+
+    def _exec_aggregation(
```

Comment:
Add a TODO: Since batch engine is assuming aggregation is an async method, we need to provide exec_aggregation_async in executor to avoid blocking.

Reply:
Added a task.